app/controllers/SiProtectedUserController.scala

```diff
@@ -16,44 +16,39 @@
 
 package controllers
 
-import _root_.config.SiProtectedUserConfig
-import controllers.actions.StrideAction
-import play.api.Logging
-import play.api.i18n.I18nSupport
+import connectors.SiProtectedUserAdminBackendConnector
+import controllers.base.{StrideAction, StrideController}
 import play.api.mvc._
-import services.SiProtectedUserListService
-import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
 import views.Views
 
 import javax.inject.{Inject, Singleton}
-import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.ExecutionContext
 
 @Singleton
-class SiProtectedUserController @Inject() (siProtectedUserConfig: SiProtectedUserConfig,
-                                           siProtectedUserListService: SiProtectedUserListService,
-                                           views: Views,
-                                           mcc: MessagesControllerComponents,
-                                           strideAction: StrideAction
-                                          )(implicit
-  ec: ExecutionContext
-) extends FrontendController(mcc)
-    with Logging
-    with I18nSupport {
+class SiProtectedUserController @Inject() (
+  val strideAction: StrideAction,
+  backendConnector: SiProtectedUserAdminBackendConnector,
+  views: Views,
+  mcc: MessagesControllerComponents
+)(implicit ec: ExecutionContext)
+    extends StrideController(mcc) {
 
-  def homepage(): Action[AnyContent] = (Action andThen strideAction)(implicit request => Ok(views.home()))
+  def homepage(filterByTeam: Option[String], searchQuery: Option[String]): Action[AnyContent] = StrideAction.async { implicit request =>
+    val teamOpt = filterByTeam filterNot "all".equalsIgnoreCase
+    val queryOpt = searchQuery filter (_.nonEmpty)
 
-  def view(entryId: String): Action[AnyContent] = (Action andThen strideAction).async { implicit request =>
-    if (!siProtectedUserConfig.shutterService) {
-      siProtectedUserListService
-        .findEntry(entryId)
-        .map {
-          case Some(protectedUser) => Ok(views.view(protectedUser))
-          case None                => NotFound(views.errorTemplate("error.not.found", "error.not.found", "protectedUser.details.not.found"))
-        }
-        .recover { case exception => InternalServerError(views.errorTemplate("error.internal_server_error", "error.internal_server_error", exception.getMessage)) }
-    } else {
-      Future.successful(Ok(views.home()))
-    }
+    backendConnector
+      .findEntries(teamOpt, queryOpt)
+      .map(entries => Ok(views.home(entries, teamOpt, queryOpt)))
   }
 
```

Comment:
I'd like to see the homepage() and view() calls going via SiProtectedUserListService as the next tickets we play on this will be to add auditing/ logging and it will be cleaner/easier to add that to the service rather than bolting more code on here

Reply:
Fair enough. I will make this change.

Reply:
either approach could work - my goal is avoid bloat in the controller class

Reply:
Thinking about it a bit - the subclass approach would become a bit unwieldy as soon as you want to do something else other than auditing as you'd create another subclass, whereas in a service class you can coordinate the ordering/interactions a little easier

Reply:
Depends on what you mean by "something else`. If it maintains the same method contract, then I think extending the class and dropping in the required implementation via the injector is a very elegant approach. If that something else does change the method contract then, yes, I would agree that the application cake requires another layer.